app.js

```diff
@@ -54,6 +59,18 @@ app.post(
   bodyParser.json({ limit: (Settings.max_doc_length + 64 * 1024) * 2 }),
   HttpController.updateDoc
 )
+app.patch(
+  '/project/:project_id/doc/:doc_id',
+  bodyParser.json(),
+  validate({
+    body: {
+      deleted: Joi.boolean(),
+      name: Joi.string().when('deleted', { is: true, then: Joi.required() }),
+      deletedAt: Joi.date().when('deleted', { is: true, then: Joi.required() })
```

Comment:
Optional: would it make sense to make `deletedAt` optional and fill in the current date if it's not supplied?

Reply:
How would we end up with a request w/o a deletedAt parameter? If that field is missing/JSON body corrupted, I'd rather reject the request entirely.

Reply:
Some future use of the API might not need to provide a deletedAt param. But it's fine to leave it as is. That was just a suggestion.